commands/operator-sdk/cmd/build.go

```diff
@@ -48,26 +48,38 @@ const (
 	build       = "./tmp/build/build.sh"
 	dockerBuild = "./tmp/build/docker_build.sh"
 	configYaml  = "./config/config.yaml"
+	watchesYaml = "./watches.yaml"
 )
 
 func buildFunc(cmd *cobra.Command, args []string) {
 	if len(args) != 1 {
 		cmdError.ExitWithError(cmdError.ExitBadArgs, fmt.Errorf("build command needs exactly 1 argument"))
 	}
 
-	bcmd := exec.Command(build)
-	o, err := bcmd.CombinedOutput()
-	if err != nil {
-		cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("failed to build: (%v)", string(o)))
+	// Don't need to buld go code if Ansible Operator
+	operatorType := determineOperatorType()
+	if operatorType == "go" {
+		bcmd := exec.Command(build)
+		o, err := bcmd.CombinedOutput()
+		if err != nil {
+			cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("failed to build: (%v)", string(o)))
+		}
+		fmt.Fprintln(os.Stdout, string(o))
 	}
-	fmt.Fprintln(os.Stdout, string(o))
 
 	image := args[0]
 	dbcmd := exec.Command(dockerBuild)
 	dbcmd.Env = append(os.Environ(), fmt.Sprintf("IMAGE=%v", image))
-	o, err = dbcmd.CombinedOutput()
+	o, err := dbcmd.CombinedOutput()
 	if err != nil {
 		cmdError.ExitWithError(cmdError.ExitError, fmt.Errorf("failed to output build image %v: (%v)", image, string(o)))
 	}
 	fmt.Fprintln(os.Stdout, string(o))
 }
+
+func determineOperatorType() string {
+	if _, err := os.Stat(watchesYaml); err == nil {
```

Comment:
On the other hand, we probably shouldn't assume that the lack of go source means ansible. Other types, like perhaps helm, may not have go source either. It could be best to positively identify one or the other, and let the default be "error; no operator type detected".

Reply:
Considering that we all that happens now, is a build of the dockerfile, that seems like the reasonable default IMO. If we add a third type we can deal with the differences then IMO.

Reply:
Need to change this so that we first check for `cmd/<name>/main.go` and if we find it we build just like we expect for a normal golang operator except with the Ansible Operator Dockerfile.